cmd/podman/root.go

```diff
@@ -114,6 +114,11 @@ func Execute() {
 			}
 		}
 		fmt.Fprintln(os.Stderr, formatError(err))
+		// In case Run errors, Cobra does not execute the PostRuns, so let's do this here explicitly.
+		// See https://github.com/spf13/cobra/issues/914.
+		if err := persistentPostRunE(rootCmd, os.Args[1:]); err != nil {
+			fmt.Fprintln(os.Stderr, formatError(err))
+		}
```

Comment:
This is not quite correct, certain commands such as machine overwrite PersistentPostRunE.
We need to actually call cmd.PersistentPostRunE() where cmd is the actual executed command.

Reply:
How do I know `cmd`?

Reply:
Yeah that is the tricky part. I do not know a good way. One possibility is to reparse the args with `rootCmd.Traverse()` but this can be expensive I guess. Maybe we can use a global var and assign `cmd` in `persistentPreRunE` to it.

Reply:
I will open an issue upstream about this.

Reply:
Can you jsut do?
```
		if err := rootCmd.PersistentPostRunE(rootCmd, os.Args[1:]); err != nil {
			fmt.Fprintln(os.Stderr, formatError(err))
		}
```